### PR TITLE
Other pages functionaties

### PR DESCRIPTION
**Pull request by prankurpandeyy**
Things to watch :+1: 

- Can add **likes** into a video and access the same video data in **likes pages** and also remove that video from **delete icon**
- Can add **watchlater** into a video and access the same video data in **watchlater** page and also remove that video from **delete icon**
- Can make **playlist** for a video and access the same playlist  in **playlist page** and also remove that video from **delete icon**
- Can track the **history** of all the videos which has been opened and watched in **history page**  also remove the whole history from **clear all history button** and remove a **single video** from **delete icon** 
- UI and code quality

### CODE REVIEW

File: slate-video-library/src/Components/Context/HistorypageContext.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  const [state, setHistoryFn] = useReducer(reducerFn, {
+    addToHistory: [],
+    getHistory: [],
+    removeCompleteHistory: [],
+    removeVideoFromHistory: [],
+  });
+
+  const { getHistory } = state;
+  console.log(
+    "🚀 ~ file: HistorypageContext.js ~ line 29 ~ HistorypageContext ~ getHistory",
```
Comment (Adarsh-Patel-Dev) [inline slate-video-library/src/Components/Context/HistorypageContext.js]: remove logs

File: slate-video-library/src/Components/Context/HistorypageContext.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    getHistory
+  );
+
+  const addToHistoryFn = async (videoData, setHistoryFn) => {
+    const response = await axios({
```
Comment (Adarsh-Patel-Dev) [inline slate-video-library/src/Components/Context/HistorypageContext.js]: you should use try catch method for error handling

File: slate-video-library/src/Components/Context/WatchlaterpageContext.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      case "GET_WATCHLATER_VIDEOS":
+        return { ...state, getWatchLaterVideos: action.payload };
+      default:
+        return state;
+    }
+  }
+
+  const [state, setWatchLaterFn] = useReducer(reducerFn, {
+    addTowatchLater: [],
+    getWatchLaterVideos: [],
```
Comment (Adarsh-Patel-Dev) [inline slate-video-library/src/Components/Context/WatchlaterpageContext.js:22]: I think you don't need 2 different arrays here
You can achieve same thing using only 1 array

File: slate-video-library/src/Pages/Likespage/Likespage.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import React, { useEffect } from "react";
+import { useLikeContext } from "../../Components/Context/LikespageContext";
+import { useSingleVideoContext } from "../../Components/Context/SingleVideopageContext";
+import Footer from "../../Components/Footer/Footer";
+import Header from "../../Components/Header/Header";
+import Sidebar from "../../Components/Sidebar/Sidebar";
+import SmallVideoCards from "../../Components/SmallVideoCards/SmallVideoCards";
```
Comment (Adarsh-Patel-Dev) [inline slate-video-library/src/Pages/Likespage/Likespage.js:7]: you can export all components in one file and then export all components from that file

**Review by Adarsh-Patel-Dev — APPROVED** [review]
Code quality has improved a lot from your last project
Good work

**Comment by varun-Tad** [comment]
Code looks good and like adarsh said I would recommend using try-catch because it helped me several times. You are good to merge.

File: slate-video-library/src/Components/Context/HistorypageContext.js (hunk 2 above)
Comment (prankurpandeyy) [reply slate-video-library/src/Components/Context/HistorypageContext.js]: done

File: slate-video-library/src/Components/Context/HistorypageContext.js (hunk 1 above)
Comment (prankurpandeyy) [reply slate-video-library/src/Components/Context/HistorypageContext.js]: done

**Pull request merged**